pkg/upgraders/healthcheck_nodes.go

```diff
@@ -0,0 +1,53 @@
+package upgraders
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/go-logr/logr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	upgradev1alpha1 "github.com/openshift/managed-upgrade-operator/api/v1alpha1"
+	"github.com/openshift/managed-upgrade-operator/pkg/machinery"
+	"github.com/openshift/managed-upgrade-operator/pkg/metrics"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func ManuallyCordonedNodes(metricsClient metrics.Metrics, machinery machinery.Machinery, c client.Client, ug *upgradev1alpha1.UpgradeConfig, logger logr.Logger) (bool, error) {
+	nodes := &corev1.NodeList{}
+	cops := &client.ListOptions{
+		Raw: &metav1.ListOptions{
+			LabelSelector: "node-role.kubernetes.io/worker, !node-role.kubernetes.io/infra",
+		},
+	}
+	// Get the list of worker nodes
+	err := c.List(context.TODO(), nodes, cops)
+	if err != nil {
+		logger.Info("Unable to fetch node list")
+		metricsClient.UpdateMetricHealthcheckFailed(ug.Name, metrics.ClusternNodesManuallyCordoned)
```

Comment:
This metric is a bit misleading. If for some reason the API itself is not available, we need to set a different reason for failure like "healthcheck_node_query_failed" like in https://github.com/openshift/managed-upgrade-operator/blob/master/pkg/upgraders/healthcheck_alerts.go#L33 for metric query failing. Here it's possible that all the kubelets in the nodes are fine but due to API issue the list of nodes doesn't work.

Mainly, the "reason" label for healthcheck metric should correctly reflect where the failure happened.

Reply:
Thank you, Ravi. Make sense, I'll make the update.